### PR TITLE
chore: 🤖 handle removed_ids from pagionation api

### PR DESCRIPTION
**Pull request by DhariniJeeva**
## Description

PR to handle removed_ids before returning the result set.

## How to Test

:technologist: [Admin preview](PATH_TO_FEATURE)

:desktop_computer: [Desktop preview](PATH_TO_FEATURE)

## Checklist:

- ~[ ] I have added before and after screenshots for UI changes~
- ~[ ] I have added JSON response output for API changes~
- ~[ ] I have added steps to reproduce and test for bug fixes in the description~
- ~[ ] I have commented on my code, particularly in hard-to-understand areas~
- ~[ ] My changes generate no new warnings~
- ~[ ] I have added tests that prove my fix is effective or that my feature works~

### CODE REVIEW

File: addons/api/addon/adapters/application.js [hunk 1]
```diff
@@ -124,9 +124,16 @@ export default class ApplicationAdapter extends RESTAdapter.extend(
       result = await super.query(store, schema, query);
       //add the result items to a data array
       if (result && result.items) {
+        //filter out the removed_ids before returning the result set
+        //This is due to items being removed from the previous iteration
+        //while the user is still paginating
```
Comment (ZedLi) [inline addons/api/addon/adapters/application.js:129]:
```suggestion
        // Filter out the removed_ids before returning the result set.
        // This is due to items being removed from the previous iteration
        // while the user is still paginating.
```

File: addons/api/addon/adapters/application.js [hunk 2]
```diff
@@ -124,9 +124,16 @@ export default class ApplicationAdapter extends RESTAdapter.extend(
...
+        if (result.removed_ids) {
+          data = data.items.filter((i) => !result.removed_ids.includes(i));
+        }
         data.push(...result.items);
         //pass in the refresh token for subsequent calls to fetch the remaining list items
         query.refresh_token = result.refresh_token;
+
         //break the loop as soon the response_type becomes complete
         if (result.response_type === 'complete') {
```
Comment (ZedLi) [inline addons/api/addon/adapters/application.js:138]: Not related to this PR but why is this `break` needed? Isn't it already handled by your while loop?

File: addons/api/addon/adapters/application.js
Comment (ZedLi) [inline addons/api/addon/adapters/application.js:1]: I seem to be getting this error while testing it, did you ever see this?

image

File: addons/api/addon/adapters/application.js [hunk 3]
```diff
@@ -124,9 +124,16 @@ export default class ApplicationAdapter extends RESTAdapter.extend(
       result = await super.query(store, schema, query);
       //add the result items to a data array
       if (result && result.items) {
+        //filter out the removed_ids before returning the result set
+        //This is due to items being removed from the previous iteration
+        //while the user is still paginating
+        if (result.removed_ids) {
+          data = data.items.filter((i) => !result.removed_ids.includes(i));
```
Comment (ZedLi) [inline addons/api/addon/adapters/application.js:131]: `data.items` doesn't exist yet.

File: addons/api/addon/adapters/application.js
Comment (ZedLi) [inline addons/api/addon/adapters/application.js:1]: It seems like the API also returns the removed IDs for each subsequent call, so we would be trying to remove an already removed item every call.

We should collate all the removed IDs in a set and then remove them from the final results at the end.

File: addons/api/addon/adapters/application.js (hunk 2 above)
Comment (DhariniJeeva) [reply addons/api/addon/adapters/application.js:138]: addressed [here](https://github.com/hashicorp/boundary-ui/pull/2032)

**Comment by DhariniJeeva** [comment]
Closing this as the API is changed and the FE doesn't need to handle`removed_ids` anymore as they are only returned when the pagination is complete and when the user clicks on the initial query again, so by then, we would only have valid resources that aren't deleted.

**Pull request closed without merging**